src/smc-webapp/app-framework/Store.ts

```diff
@@ -135,7 +139,7 @@ export class Store<State> extends EventEmitter {
    * happens call the given callback.
    */
   wait<T>(opts: {
-    until: (store: Store<State>) => T; // waits until "until(store)" evaluates to something truthy
+    until: (store: any) => T; // waits until "until(store)" evaluates to something truthy
```

Comment:
Typescript's inability to write `(store: this) => T` as a function argument where `this` is either a Store or subclass makes this awkward either way. Callers of `wait` should specify the store instance type.

Reply:
Why isn't this an actual comment in the source code?  It seems pretty useful for people using the code...